library/core/src/num/int_macros.rs

```diff
@@ -1581,7 +1581,37 @@ macro_rules! int_impl {
             if self < 0 {
                 None
             } else {
-                Some((self as $UnsignedT).isqrt() as Self)
+                // SAFETY: Input is nonnegative in this `else` branch.
+                let result = unsafe {
+                    crate::num::int_sqrt::$ActualT(self as $ActualT) as $SelfT
+                };
+
+                // SAFETY: Inform the optimizer what the range of outputs is.
+                //
+                // Integer square root is a monotonically nondecreasing
```

Comment:
Nit: informing the optimizer is only the action being taken, the rest is what is actually needed to meet the precondition (applies a few other places too)

```suggestion
                // Inform the optimizer what the range of outputs is.
                //
                // SAFETY: Integer square root is a monotonically nondecreasing ...
```

Reply:
OK, I've changed that.